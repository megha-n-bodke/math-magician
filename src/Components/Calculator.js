import React from 'react';
import calculate from '../logic/calculate';

class Calculator extends React.Component {
  constructor(props) {
    super(props);

    this.state = { total: 0, operation: null, next: null };
  }

  clickHandler = (e) => {
    this.setState((state) => calculate(state, e.target.textContent));
  };

  render() {
    const { total, next, operation } = this.state;
    return (
      <div className="calculator-container">
        <table className="calculator">
          <tr className="answer-row">
            <td colSpan="4">
              {total}
              {operation}
              {next}
            </td>
          </tr>

          <tr>
            <td onClick={this.clickHandler}>AC</td>
            <td onClick={this.clickHandler}>+/-</td>
            <td onClick={this.clickHandler}>%</td>
            <td className="operator-column" onClick={this.clickHandler}>
              &#247;
            </td>
          </tr>

          <tr>
            <td onClick={this.clickHandler}>7</td>
            <td onClick={this.clickHandler}>8</td>
            <td onClick={this.clickHandler}>9</td>
            <td className="operator-column" onClick={this.clickHandler}>
              x
            </td>
          </tr>

          <tr>
            <td onClick={this.clickHandler}>4</td>
            <td onClick={this.clickHandler}>5</td>
            <td onClick={this.clickHandler}>6</td>
            <td className="operator-column" onClick={this.clickHandler}>
              -
            </td>
          </tr>

          <tr>
            <td onClick={this.clickHandler}>1</td>
            <td onClick={this.clickHandler}>2</td>
            <td onClick={this.clickHandler}>3</td>
            <td className="operator-column" onClick={this.clickHandler}>
              +
            </td>
          </tr>

          <tr>
            <td colSpan={2} onClick={this.clickHandler}>
              0
            </td>

            <td onClick={this.clickHandler}>.</td>
            <td className="operator-column" onClick={this.clickHandler}>
              =
            </td>
          </tr>
        </table>
      </div>
    );
  }
}

export default Calculator;
