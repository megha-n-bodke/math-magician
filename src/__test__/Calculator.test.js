import renderer from 'react-test-renderer';
import Calculator from '../Components/Calculator';

test('Calculator snapshot testing', () => {
  const navbar = renderer.create(<Calculator />);
  const navObject = navbar.toJSON();
  expect(navObject).toMatchSnapshot();
});
