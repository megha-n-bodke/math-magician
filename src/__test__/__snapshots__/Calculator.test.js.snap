// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator snapshot testing 1`] = `
<div
  className="calciPage"
>
  <p
    className="calciPara"
  >
    Lets Do Some Math!!
  </p>
  <div
    className="calculator-container"
  >
    <table
      className="calculator"
    >
      <tr
        className="answer-row"
      >
        <td
          colSpan="4"
        />
      </tr>
      <tr>
        <td
          onClick={[Function]}
        >
          AC
        </td>
        <td
          onClick={[Function]}
        >
          +/-
        </td>
        <td
          onClick={[Function]}
        >
          %
        </td>
        <td
          className="operator-column"
          onClick={[Function]}
        >
          รท
        </td>
      </tr>
      <tr>
        <td
          onClick={[Function]}
        >
          7
        </td>
        <td
          onClick={[Function]}
        >
          8
        </td>
        <td
          onClick={[Function]}
        >
          9
        </td>
        <td
          className="operator-column"
          onClick={[Function]}
        >
          x
        </td>
      </tr>
      <tr>
        <td
          onClick={[Function]}
        >
          4
        </td>
        <td
          onClick={[Function]}
        >
          5
        </td>
        <td
          onClick={[Function]}
        >
          6
        </td>
        <td
          className="operator-column"
          onClick={[Function]}
        >
          -
        </td>
      </tr>
      <tr>
        <td
          onClick={[Function]}
        >
          1
        </td>
        <td
          onClick={[Function]}
        >
          2
        </td>
        <td
          onClick={[Function]}
        >
          3
        </td>
        <td
          className="operator-column"
          onClick={[Function]}
        >
          +
        </td>
      </tr>
      <tr>
        <td
          colSpan={2}
          onClick={[Function]}
        >
          0
        </td>
        <td
          onClick={[Function]}
        >
          .
        </td>
        <td
          className="operator-column"
          onClick={[Function]}
        >
          =
        </td>
      </tr>
    </table>
  </div>
</div>
`;
